Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -50,11 +50,12 @@ open class _ChatChannelListVC<ExtraData: ExtraDataTypes>: _ViewController,
         controller.synchronize()
         
         collectionView.register(uiConfig.channelList.collectionViewCell.self, forCellWithReuseIdentifier: "Cell")
-
-        if let cellSeparatorIdentifier = (collectionViewLayout as? ListCollectionViewLayout)?.separatorIdentifier {
-            collectionViewLayout.register(
+        
+        if let cellSeparatorKind = (collectionViewLayout as? ListCollectionViewLayout)?.separatorKind {
+            collectionView.register(
                 uiConfig.channelList.cellSeparatorReusableView,
-                forDecorationViewOfKind: cellSeparatorIdentifier
+                forSupplementaryViewOfKind: cellSeparatorKind,
+                withReuseIdentifier: "CellSeparatorIdentifier"
             )
         }
```

Comment:
Hmmm, let's do it the same way UICollectionView does it with section headers and footers:
- make the separator kind a class var `ListCollectionViewLayout.separatorKind`
- the reuse identifier could either be a local constant used within the VC or we could use `uiConfig.channelList.cellSeparatorReusableView.reuseIdentifier`

Reply:
yep, the `static let` sounds better 🙌 

as for the `reuseIdentifier`, do we want people to access it? afaict it's the same case as for the cell's reuse identifier, so I'd be for keeping it local

Reply:
I think we don't need to specify the reuse identifier locally by default. It's ok to use `uiConfig.channelList.cellSeparatorReusableView.reuseIdentifier` here too, isn't it?

Reply:
we can even remove the `if` check since you can register supplementary views as you want. worst case is that the layout won't use them

Reply:
sure, we can do it like that)